src/generic/canbus.c

```diff
@@ -153,6 +161,21 @@ can_process_set_nodeid(uint32_t id, uint32_t len, uint8_t *data)
     }
 }
 
+static void
+can_process_request_bootloader(uint32_t id, uint32_t len, uint8_t *data)
+{
+    if (!can_check_uuid(id, len, data))
+        return;
+    irq_disable();
```

Comment:
This `irq_disable()` should be right before the `*req_sig = ...` line.  Otherwise, if the canboot bootloader is not found then the function will return with irqs disabled which could cause havoc.  Only the assignment and reset needs to be protected.

Reply:
Oops, that would be bad.  Fixed.